src/linux/BTSerialPortBindingServer.cc

```diff
@@ -385,65 +401,70 @@ void BTSerialPortBindingServer::Advertise(listen_baton_t * baton) {
 
     uint8_t rfcomm_channel = (uint8_t) baton->listeningChannelID;
 
-    const char *service_name = "RFCOMM Server socket";
-    const char *service_dsc = "A RFCOMM listening socket";
-    const char *service_prov = "Service Provider";
+    const char *service_name = "Serial Port";
+    const char *service_dsc = "An RFCOMM listening socket";
 
-    uuid_t root_uuid, l2cap_uuid, rfcomm_uuid;
-    sdp_list_t *l2cap_list = 0, 
+    uuid_t root_uuid, service_uuid, l2cap_uuid, rfcomm_uuid;
+    sdp_list_t *l2cap_list = 0,
                *rfcomm_list = 0,
                *root_list = 0,
-               *proto_list = 0, 
+               *proto_list = 0,
+               *service_class_list = 0,
                *access_proto_list = 0;
     sdp_data_t *channel = 0;
 
     sdp_record_t *record = sdp_record_alloc();
 
     sdp_set_service_id(record, baton->uuid);
 
+    sdp_uuid16_create(&service_uuid, _SPP_UUID);
```

Comment:
Shouldn't the `service_uuid` be passed in as a parameter? Will the UUID that passed via the `options` object to the javascript `listen` be advertised? Or am I missing something?